ocfweb/lab_reservations/reserve.py

```diff
@@ -0,0 +1,178 @@
+from collections import namedtuple
+
+import ocflib.misc.validators
+from django import forms
+from django.shortcuts import render
+from jinja2 import Environment
+from jinja2 import PackageLoader
+from ocflib.account.search import user_attrs_ucb
+from ocflib.misc.mail import send_mail
+
+from ocfweb.auth import calnet_required
+from ocfweb.component.forms import wrap_validator
+
+
+JINJA_MAIL_ENV = Environment(loader=PackageLoader('ocfweb', ''))
+
+
+class NewReservationRequest(namedtuple(
+    'NewReservationRequest', [
+        'real_name',
+        'contact_email',
+        'student_group',
+        'reason',
+        'date',
+        'starttime',
+        'endtime',
+    ],
+)):
+    """Request for reservation request.
+    :param real_name:
+    :param contact_email:
+    :param student_group:
+    :param reason:
+    :param date:
+    :param starttime:
+    :param endtime:
+    """
+
+    def to_dict(self):
+        return {
+            field: getattr(self, field)
+            for field in self._fields
+        }
```

Comment:
There's a built-in `_asdict()` method on namedtuples you can use:

```python
>>> import collections
>>> Point = collections.namedtuple('Point', ('x', 'y'))
>>> Point(x=32, y=16)._asdict()
OrderedDict([('x', 32), ('y', 16)])
```

It should be possible to just call the `namedtuple` constructor here without inheriting from it:

```python
NewReservationRequest = namedtuple(
    'NewReservationRequest',
    (
        'real_name',
        'contact_email',
        'student_group',
        'reason',
        'date',
        'starttime',
        'endtime',
    ),
)
```

I don't think we're really losing anything by losing the docstring, since it just repeats the parameters. Plus if you do this, you can ignore my slots comment above.